### ISSUE TITLE
Support for submodules

### ISSUE THREAD

**Issue by piotrb** [Arkweid/lefthook#38]
When running lefthook on a checked out submodule the directory structure is not as expected.

```
$ lefthook install
SYNCING lefthook.yml
SERVED HOOKS: pre-commit, prepare-commit-msg
open {HOME}/{PROJECT}/{SUBMODULE_NAME}/.git/hooks/pre-commit: not a directory
```

This is because in a submodule `.git` is a file with the following content:
```
gitdir: ../.git/modules/{SUBMODULE_NAME}
```

It sounds simple enough to have lefthook understand that and just follow that relative path instead and drop the hooks there.

(I did a quick test, that's all that's needed, if you drop the hooks in that folder instead it seems to work as expected)

**Comment by Arkweid** [comment]
Thank you for issue @piotrb!
Unfortunately, I'm a little busy in the coming days :(
If you have the opportunity to do PR with a correction, this will be great. I will accept it and release with your fix.

**Comment by Lighttree** [comment]
Hi I have one issue with my project structure.
And I'm not sure is it in scope of this `enhancement` or not.

I'm trying to migrate `JavaScript` project from `Husky`. It has following folder structure:
```
/ my_project_parent
  /.git
  /  my_project
     lefthook.yml
     package.json
     ...
...
```

The problem is that lefthook looking for `.git` folder in `my_project` at the moment, while it is actually in parent folder.

It would be nice if it can find parent `.git` and write `hooks` there.

**Comment by Arkweid** [comment]
Yea, just need to replace calls form `current_working_directory` to result of that call: `git rev-parse --show-toplevel`

**Comment by lisovskyvlad** [comment]
Would be awesome as next feature!

**Comment by Arkweid** [comment]
Released in 0.7.0
More info here:
https://github.com/Arkweid/lefthook/issues/72
